manifests/init.pp

```diff
@@ -7,10 +7,13 @@
 # $port::             port of foreman proxy that is used as $server
 #
 # $ca_file::          path to file of certification authority that issued client's certificate
+#                     May be overriden if $::rh_certificate_repo_ca_file (from Facter) is found
```

Comment:
I think param class does not help here much as we have 2 sources of default value here. Either we have `$::rh_certificate_repo_ca_file` set (custom fact or a global parameter) or we want to have puppet cert as default value. But others especially those puppet gurus might have better answer.